src/util/options.rs

```diff
@@ -426,6 +426,165 @@ impl Options {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum GCTriggerSelector {
+    FixedHeapSize(usize),
+    DynamicHeapSize(usize, usize),
+    Delegated,
+}
+
+impl GCTriggerSelector {
+    const K: usize = 1024;
+    const M: usize = 1024 * 1024;
+    const G: usize = 1024 * 1024 * 1024;
+
+    /// Parse a size representation, which could be a number to represents bytes,
+    /// or a number with the suffix K/k/M/m/G/g. Return the byte number if it can be
+    /// parsed properly, otherwise return an error string.
+    fn parse_size(s: &str) -> Result<usize, String> {
+        println!("parse: {:?}", s);
+        let s = s.to_lowercase();
+        if s.ends_with(char::is_alphabetic) {
+            let num = s[0..s.len() - 1]
+                .parse::<usize>()
+                .map_err(|e| e.to_string())?;
+            if s.ends_with('k') {
+                Ok(num * Self::K)
+            } else if s.ends_with('m') {
+                Ok(num * Self::M)
+            } else if s.ends_with('g') {
+                Ok(num * Self::G)
```

Comment:
Even if `num` is within the range of `usize`, this may still overflow.  For example, "1073741824G" is equal to 2 to the power of 64, and that overflows usize.  For 32-bit machines, "4G" already overflows `usize`.

Use `usize::checked_mul` instead.  Alternatively, always use `u64` or `u128` when computing, and convert back to `usize` afterwards.

Reply:
Done. I am parse the number and do multiplication in `u64`, and convert to usize at the end.